### PR TITLE
refactor(workflow): separate workflow with filter for backend testing

### PR DESCRIPTION
**Pull request by mahula**
## 🍰 Pullrequest
Separate the test workflow for backend.

- [x] add file filter
- [x] transfer all jobs from general `test.yml` to separate workflow file `test_backend.yml`
- [x] use  the file change check job in `test_backend.yml`
- [x] use file change check results as condition for job runs
- [ ] adapt check requirements for master branch protection (@ulfgebhardt)
      - set the workflow jobs `Detect File Changes - Backend` and `Unit Tests - Backend` as required
      - remove `Unit tests - Backend` from the list of required checks

### Issues
- relates #2664

### CODE REVIEW

File: .github/workflows/test_backend.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      - name: nginx | Build 'test' image
+        run: |
+          docker build -t "gradido/nginx:test" nginx/
+          docker save "gradido/nginx:test" > /tmp/nginx.tar
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: docker-nginx-test
+          path: /tmp/nginx.tar
```
Comment (ulfgebhardt) [inline .github/workflows/test_backend.yml:103]: this has to be in a nginx workflow

File: .github/workflows/test_backend.yml [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    name: Lint - Database Up
+    if: needs.files-changed.outputs.backend == 'true'
+    needs: files-changed
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Database | Lint
+        run: cd database && yarn && yarn run lint
```
[the remark this answers is not in the document]
Comment (ulfgebhardt) [reply .github/workflows/test_backend.yml:139]: Database workflow has to run for backend, dht-node & federation

File: .github/workflows/test_frontend.yml [hunk 3]
```diff
@@ -23,7 +24,7 @@ jobs:
 
   
   build_test:
-    if: needs.files-changed.outputs.frontend == 'true'
+    if: needs.files-changed.outputs.docker == 'true' || needs.files-changed.outputs.frontend == 'true'
```
Comment (ulfgebhardt) [inline .github/workflows/test_frontend.yml:27]: not needed

**Review by ulfgebhardt — COMMENTED** [review]
DHT + Federation + backend -> dependent on database
Database -> dependent on mariadb
all workflows who use `docker-compose` (not `docker`) -> dependent on docker-compose file changes

**Comment by mahula** [comment]
This PR is obsolete and can be closed.
All issues where covered in #2894 and #2962.

**Pull request closed without merging**